source.go

```diff
@@ -0,0 +1,156 @@
+package conf
+
+import (
+	"bytes"
+	"flag"
+	"html/template"
+	"reflect"
+	"strings"
+)
+
+// Source is the interface that allow new types to be plugged into a loader to
+// make it possible to load configuration from new places.
+//
+// When the configuration is loaded the Load method of each source that was set
+// on a loader is called with a pointer to an in-memory structure representing
+// the configuration object passed as dst.
+type Source interface {
+	Load(dst interface{}) error
+}
+
+// FlagSource is a special case of a source that receives a configuration value
```

Comment:
What i htink would be useful is overriding `argv` for flags passed to the loader... then you could support things like double dash, non-segmentio/conf args, etc.